### PR TITLE
Update github actions to not use deprecated node 16

### LINKED ISSUE #42617: [Improvement]: Update used github actions to not use deprecated node 16
### Description

Many pipelines still use versions of actions that are using the deprecated node version 16:
image
 

### Describe your problem(s)

_No response_

### Describe your solution(s)

Update the actions used in the pipelines

### Related area

-> CICD

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

I would like to contribute myself

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
Update github actions to not use deprecated node 16.

Fixes #42617

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by Shadow-Devil** [comment]
Could someone please review and merge this? Thank you in advance!

**Review by ThisaruGuruge — APPROVED** [review]

**Comment by ThisaruGuruge** [comment]
> Could someone please review and merge this? Thank you in advance!

Thank you for your contribution, @Shadow-Devil, and we are sorry for the late response. I have approved the PR, let's wait until the checks pass, and the I will merge this. Hoping to see more contributions from you in the future.

**Pull request merged**
